core/optaplanner-core-impl/src/main/java/org/optaplanner/core/impl/domain/solution/descriptor/SolutionDescriptor.java

```diff
@@ -112,7 +112,7 @@
     public static <Solution_> SolutionDescriptor<Solution_> buildSolutionDescriptor(DomainAccessType domainAccessType,
             Class<Solution_> solutionClass, Map<String, MemberAccessor> memberAccessorMap,
             Map<String, SolutionCloner> solutionClonerMap, List<Class<?>> entityClassList) {
-        memberAccessorMap = Objects.requireNonNullElse(memberAccessorMap, Collections.emptyMap());
+        memberAccessorMap = Objects.requireNonNullElse(memberAccessorMap, new HashMap<>(0));
```

Comment:
Initializing with size 0 is a bad idea. Let's just not specify the size (which will give 16 which is extremely low memory usage)?

```suggestion
        memberAccessorMap = Objects.requireNonNullElse(memberAccessorMap, new HashMap<>());
```